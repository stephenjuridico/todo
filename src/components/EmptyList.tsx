import React from 'react';
import {Center, Text} from 'native-base';

interface EmptyListProps {
  message: string;
}

const EmptyList: React.FC<EmptyListProps> = ({message}) => {
  return (
    <Center minH={200} flex={1}>
      <Text
        testID="empty-list-message"
        width={'80%'}
        textAlign={'center'}
        color={'gray.400'}>
        {message}
      </Text>
    </Center>
  );
};

export default EmptyList;
