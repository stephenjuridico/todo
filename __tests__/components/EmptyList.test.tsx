import 'react-native';
import React from 'react';
import EmptyList from '../../src/components/EmptyList';
import {render, screen} from '../../src/test/test-utils';

describe('EmptyList', () => {
  const expectedMessage = 'Empty List';
  const testId = 'empty-list-message';

  it('renders correctly', () => {
    render(<EmptyList message={expectedMessage} />);
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('renders correctly and displays message', () => {
    render(<EmptyList message={expectedMessage} />);
    expect(screen.getByTestId(testId).children).toContain(expectedMessage);
  });
});
