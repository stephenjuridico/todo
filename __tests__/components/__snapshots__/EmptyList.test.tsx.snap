// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyList renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={{}}
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flex": 1,
        "justifyContent": "center",
        "minHeight": 200,
      }
    }
  >
    <Text
      dataSet={{}}
      style={
        {
          "backgroundColor": undefined,
          "color": "#a1a1aa",
          "fontFamily": undefined,
          "fontSize": 14,
          "fontStyle": "normal",
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 21,
          "textAlign": "center",
          "textDecorationLine": undefined,
          "width": "80%",
        }
      }
      testID="empty-list-message"
    >
      Empty List
    </Text>
  </View>
</RNCSafeAreaProvider>
`;
